rmf_robot_sim_common/src/slotcar_common.cpp

```diff
@@ -223,8 +223,10 @@ void SlotcarCommon::handle_diff_drive_path_request(
 
     Eigen::Quaterniond quat(
       Eigen::AngleAxisd(msg->path[i].yaw, Eigen::Vector3d::UnitZ()));
-    trajectory.at(i).translation() = v3;
-    trajectory.at(i).linear() = Eigen::Matrix3d(quat);
+    trajectory.at(i).pose.translation() = v3;
+    trajectory.at(i).pose.linear() = Eigen::Matrix3d(quat);
+    if (msg->path[i].obey_approach_speed_limit)
+      trajectory.at(i).approach_speed_limit = msg->path[i].approach_speed_limit;
```

Comment:
`approach_speed_limit` should be check that it is larger than 0, as it looks like it is not checked further down the code path, https://github.com/open-rmf/rmf_simulation/blob/feature/speed_limits/rmf_robot_sim_common/src/utils.cpp#L32